templates/task/cf.yml

```diff
@@ -51,19 +55,9 @@ Resources:
       NetworkMode: awsvpc
       Cpu: !Ref TaskCPU
       Memory: !Ref TaskMemory
-      ExecutionRoleArn: !Ref ExecutionRole
+      ExecutionRoleArn: !If [HasExecutionRole, !Ref ExecutionRole, !Ref "AWS::NoValue"]
```

Comment:
I wonder if our default should be to create an execution role with the same permissions as `AmazonECSTaskExecutionRolePolicy`: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_execution_IAM_role.html what do you think? Otherwise they won't even be able to pull the image when empty so I don't think it's a great default behavior.

Reply:
You are right. I actually created a default execution role but I forgot to use it...